docs/asciidoc/static/include/pluginbody.asciidoc

```diff
@@ -0,0 +1,1145 @@
+
+**Version messaging from Logstash**
+
+If you start Logstash with the `--verbose` flag, you will see messages like
+these to indicate the relative maturity indicated by the plugin version number:
+
+** **0.1.x**
++
+-----
+This plugin isn't well supported by the community and likely has no maintainer.
+-----
+
+** **0.9.x**
++
+-----
+This plugin should work but would benefit from use by folks like you. Please let us know if you find bugs or have suggestions on how to improve this plugin.
+-----
+
+** **1.x.x**
+You will no longer see a message indicating potential code immaturity when a
+plugin reaches version 1.0.0
+
+==== Runtime & Development Dependencies
+At the bottom of the `gemspec` file is a section with a comment:
+`Gem dependencies`.  This is where any other needed gems must be mentioned. If
+a gem is necessary for your plugin to function, it is a runtime dependency. If
+a gem are only used for testing, then it would be a development dependency.
+
+[NOTE]
+=========================
+You can also have versioning requirements for your dependencies--including other
+Logstash plugins:
+
+[source,ruby]
+[subs="attributes"]
+----------------------------------
+  # Gem dependencies
+  s.add_runtime_dependency 'logstash', '>= 1.4.0', '< 2.0.0'
+  s.add_development_dependency 'logstash-devutils'
+----------------------------------
+This gemspec has a runtime dependency on the core Logstash gem and requires that
+it have a version number greater than or equal to version 1.4.0 `'>= 1.4.0'`,
+and less than version 2.0 `'< 2.0.0'`.
+=========================
+
+
+IMPORTANT: All plugins have a runtime dependency on the `logstash` core gem, and
+a development dependency on `logstash-devutils`.
+
+=== Add Tests
+Logstash loves tests. Lots of tests. If you're using your new {plugintype}
+plugin in a production environment, you'll want to have some tests to ensure you
+are not breaking any existing functionality.
+
+NOTE: A full exposition on RSpec is outside the scope of this document. Learn
+more about RSpec at http://rspec.info
+
+For help learning about tests and testing, look in the
+`spec/pass:attributes[{plugintype}]s/` directory of several other similar
+plugins.
+
+=== Clone and test!
+Now let's start with a fresh clone of the plugin, build it and run the tests.
+
+* **Clone your plugin into a temporary location** Replace `GITUSERNAME` with
+your github username, and `MYPLUGINNAME` with your plugin name.
+** `git clone https://github.com/GITUSERNAME/logstash-{plugintype}-MYPLUGINNAME.git`
+*** alternately, via ssh: `git clone git@github.com:GITUSERNAME/logstash-{plugintype}-MYPLUGINNAME.git`
+** `cd logstash-{plugintype}-MYPLUGINNAME`
+
+Then, you'll need to install your plugins dependencies with bundler:
+[source,sh]
+----------------------------------
+bundle install
+----------------------------------
+
+And finally, run the tests:
+[source,sh]
+----------------------------------
+bundle exec rspec
+----------------------------------
```

Comment:
teeny tiny nit: this highlights `exec` as it is an actual system command. Could maybe lose the `[source,sh]` bit here?